apps/antalmanac/src/components/RightPane/Map/UCIMap.tsx

```diff
@@ -461,6 +545,50 @@ export default class UCIMap extends PureComponent {
                 );
             }
         }
+        for (const buildingCode in customEventPins) {
```

Comment:
In the same vein as my comment earlier in this file, I think a lot of this stuff is the same for custom and normal events so we should reuse more code. Is there any way you can reduce the diff here so we aren't adding 40 more lines to deal with custom event logic?

Reply:
For this file, it is a bit hard to do so. Regular class pin contains regular `locationData` and detailed information. But for custom event, it has special detailed information to show (for example, no `room` info and the it is called `event` rather then `class`). Also the calculation of locationData for customevent is bit different, because some location for custom event has not `()`. so the formula for calculate regular class's locationData is not applicable here. So I will still remain two part like that.